### PR TITLE
Global search is barely visible to users

### LINKED ISSUE #1382: Global search is barely visible to users
The icon is not contrasty enough in light theme 
image

also users seem to be unaware of that option

Solution: Keep searchbar open all the time

N.B. Keep the hotkey to focus fast

### PR DESCRIPTION
**Pull request by awinogradov**
## PR includes

- [x] Refactor
- [x] Feature

## Related issues

Resolve #1382

Related issues #914, #1568

## QA Instructions, Screenshots, Recordings
  

https://github.com/taskany-inc/issues/assets/982072/88bf7d19-b688-454e-869f-f858860c23c3

### CODE REVIEW

File: src/components/GoalCriteria/GoalCriteria.tsx [hunk 1]
```diff
@@ -209,11 +209,11 @@ const GoalCriteriaItem: React.FC<GoalCriteriaItemProps> = (props) => {
     }, [item]);
 
     return (
-        <StyledTableRow
+        <StyledGoalListItemCompact
             actions={availableActions}
             onActionClick={handleChange}
             item={itemToRender}
-            rowIcon={
+            rawIcon={
```
Comment (LamaEats) [inline src/components/GoalCriteria/GoalCriteria.tsx:216]: why are you rename this prop? it's icon for table row not raw :)

File: src/components/GoalListItemCompact.tsx [hunk 2]
```diff
@@ -168,29 +161,22 @@ const Column: ColumnRender = ({ col, componentProps }) => {
...
-            align={align}
-            justify={justify}
-        >
-            <StyledCell key="icon" forIcon min>
-                {rowIcon || <GoalIcon size="s" />}
-            </StyledCell>
+        <TableRow interactive={attrs.focused != null} gap={gap} align={align} justify={justify} {...attrs}>
+            {nullable(icon || rawIcon, () => (
+                <StyledCell key="icon" forIcon min>
+                    {rawIcon || <IconTargetOutline size="s" />}
```
Comment (LamaEats) [inline src/components/GoalListItemCompact.tsx:177]: can pass `<IconTargetOutline size="s" />` as default value for `rawIcon` prop

**Comment by awinogradov** [comment]
ping @9teen90nine

**Review by 9teen90nine — APPROVED** [review]

File: src/components/ListView.tsx [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+export const ListView: React.FC<ListViewProps> = ({ children, onKeyboardClick }) => {
+    const downPress = useKeyPress('ArrowDown');
+    const upPress = useKeyPress('ArrowUp');
+    const spacePress = useKeyPress(' ');
+    const enterPress = useKeyPress('Enter');
+    const [cursor, setCursor] = useState<number | undefined>();
+    const [hovered, setHovered] = useState<number | undefined>();
+    const items: unknown[] = useMemo(() => [], []);
+
+    const registerItem = useCallback((value: unknown) => items.push(value) - 1, [items]);
```
Comment (LamaEats) [inline src/components/ListView.tsx:44]: why you are changing memoized variable instead of `useState` returned value through setter function?

**Review by LamaEats — CHANGES_REQUESTED** [review]

File: src/components/ListView.tsx (hunk 3 above)
Comment (LamaEats) [reply src/components/ListView.tsx:44]: or use returned value from `useRef` function

Comment (LamaEats) [reply src/components/ListView.tsx:44]: also this code is not safety for multiple re-renders of the `ListViewItem` component

**Pull request merged**